tools/src/main/python/indexer.py

```diff
@@ -47,40 +47,82 @@ def __init__(self, command, logger=None, java=None, jar='opengrok.jar',
                  java_opts=None, env_vars=None):
 
         java_options = []
-        java_options.extend(self.get_SCM_properties(logger))
         if java_opts:
             java_options.extend(java_opts)
+        java_options = merge_properties(java_options,
+                                        get_SCM_properties(logger))
         logger.debug("Java options: {}".format(java_options))
 
         super().__init__(command, jar=jar, java=java, java_opts=java_options,
                          logger=logger, env_vars=env_vars)
 
-    def get_SCM_properties(self, logger):
-        """
-        Return list of Java System properties that contain valid paths to
-        SCM commands.
-        """
-        SCM_COMMANDS = {
-            'bk': '-Dorg.opengrok.indexer.history.BitKeeper',
-            'hg': '-Dorg.opengrok.indexer.history.Mercurial',
-            'cvs': '-Dorg.opengrok.indexer.history.cvs',
-            'svn': '-Dorg.opengrok.indexer.history.Subversion',
-            'sccs': '-Dorg.opengrok.indexer.history.SCCS',
-            'cleartool': '-Dorg.opengrok.indexer.history.ClearCase',
-            'git': '-Dorg.opengrok.indexer.history.git',
-            'p4': '-Dorg.opengrok.indexer.history.Perforce',
-            'mtn': '-Dorg.opengrok.indexer.history.Monotone',
-            'blame': '-Dorg.opengrok.indexer.history.RCS',
-            'bzr': '-Dorg.opengrok.indexer.history.Bazaar'}
-
-        properties = []
-        for cmd in SCM_COMMANDS.keys():
-            executable = get_command(logger, None, cmd, level=logging.INFO)
-            if executable:
-                properties.append("{}={}".
-                                  format(SCM_COMMANDS[cmd], executable))
-
-        return properties
+
+def get_SCM_properties(logger):
+    """
+    Return list of Java System properties that contain valid paths to
+    SCM commands.
+    """
+    SCM_COMMANDS = {
+        'bk': '-Dorg.opengrok.indexer.history.BitKeeper',
+        'hg': '-Dorg.opengrok.indexer.history.Mercurial',
+        'cvs': '-Dorg.opengrok.indexer.history.cvs',
+        'svn': '-Dorg.opengrok.indexer.history.Subversion',
+        'sccs': '-Dorg.opengrok.indexer.history.SCCS',
+        'cleartool': '-Dorg.opengrok.indexer.history.ClearCase',
+        'git': '-Dorg.opengrok.indexer.history.git',
+        'p4': '-Dorg.opengrok.indexer.history.Perforce',
+        'mtn': '-Dorg.opengrok.indexer.history.Monotone',
+        'blame': '-Dorg.opengrok.indexer.history.RCS',
+        'bzr': '-Dorg.opengrok.indexer.history.Bazaar'}
+
+    properties = []
+    for cmd in SCM_COMMANDS.keys():
+        executable = get_command(logger, None, cmd, level=logging.INFO)
+        if executable:
+            properties.append("{}={}".
+                              format(SCM_COMMANDS[cmd], executable))
+
+    return properties
+
+
+def merge_properties(base, extra):
```

Comment:
I'd introduce a 'name' cache - should probably be a set, see. https://stackoverflow.com/questions/2831212/python-sets-vs-lists - (i.e. the first element of the item.split("=")) and then I'd process 'base' first (populating the cache along the way) and then proceed with processing 'extra' depending on whether given name is in the cache or not.